seed/helpers.go

```diff
@@ -100,3 +101,11 @@ func readInfo(snapPath string, si *snap.SideInfo) (*snap.Info, error) {
 	}
 	return snap.ReadInfoFromSnapFile(snapf, si)
 }
+
+func readGadgetInfo(snapPath string, constraints *gadget.ModelConstraints) (*gadget.Info, error) {
+	snapf, err := snap.Open(snapPath)
```

Comment:
Open is not expensive but a little bit,  in theory we could teach readInfo to return the snapf on success, then we don't need this extra helper